### ISSUE TITLE
Standard Transformer Encoder/Decoder Layer

### ISSUE THREAD

**Issue by kamalkraj** [tensorflow/addons#2440]
**Describe the feature and the current behavior/state.**

**Relevant information**
- Are you willing to contribute it (yes/no):yes
*If you wish to contribute, then read the requirements for new contributions in [`CONTRIBUTING.md`](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#requirements-for-new-contributions-to-the-repository)*
- Are you willing to maintain it going forward? (yes/no):yes
- Is there a relevant academic paper? (if so, where):https://arxiv.org/abs/1706.03762
- Does the relavent academic paper exceed 50 citations? (yes/no):yes
- Is there already an implementation in another framework? (if so, where):https://pytorch.org/docs/stable/nn.html#transformer-layers
- Was it part of tf.contrib? (if so, where):No

**Which API type would this fall under (layer, metric, optimizer, etc.)**
Layer

**Who will benefit with this feature?**
Anyone who wanted to implement transformer based model (BERT, ALBERT, RoBERTa...etc) 

**Any other info.**

**Comment by fsx950223** [comment]
It's implemented in https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/layers/transformer_encoder_block.py
